pkg/arvo/app/metadata-hook.hoon

```diff
@@ -0,0 +1,200 @@
+  ::
+      %remove
+    =/  ship  (~(get by synced) path.act)
+    ?~  ship  [~ state]
+    ?:  &(?!(=(u.ship our.bowl)) ?!((team:title our.bowl src.bowl)))
+      [~ state]
+    :_  state(synced (~(del by synced) path.act))
+    %-  zing
+    :~  (unsubscribe [%group path.act] u.ship)
+        ?.  &(=(u.ship our.bowl) (team:title our.bowl src.bowl))  ~
+        [%give %kick ~[[%group path.act]] ~]~
+    ==
+  ==
+  ::
+  ++  unsubscribe
+    |=  [=path =ship]
+    ^-  (list card)
+    ?:  =(ship our.bowl)
+      [%pass path %agent [our.bowl %metadata-store] %leave ~]~
+    [%pass path %agent [ship %metadata-hook] %leave ~]~
+  --
+::
+++  watch-group
+  |=  =path
+  ^-  (list card)
+  |^
+  ?>  (~(has by synced) path)
+  ?>  (is-permitted [(scot %p src.bowl) path])
+  %+  turn  ~(tap by (metadata-scry path))
+  |=  [[=group-path =resource] =metadata]
+  ^-  card
+  [%give %fact ~ %metadata-update !>([%add group-path resource metadata])]
+  ::
+  ++  is-permitted
+    |=  pax=^path
+    ^-  ?
+    =.  pax
+      ;:(weld /=permission-store/(scot %da now.bowl)/permitted pax /noun)
+    .^(? %gx pax)
+  ::
+  ++  metadata-scry
+    |=  pax=^path
+    ^-  associations
+    =.  pax  ;:(weld /=metadata-store/(scot %da now.bowl)/group pax /noun)
+    (need .^((unit associations) %gx pax))
+  --
+::
+++  fact-metadata-update
+  |=  [wir=wire fact=metadata-update]
+  ^-  (quip card _state)
+  |^
+  [?:((team:title our.bowl src.bowl) handle-local handle-foreign) state]
+  ::
+  ++  handle-local
+    ?+  -.fact  ~
+        %add
+      ?.  (~(has by synced) group-path.fact)  ~
+      (give group-path.fact fact)
+    ::
+        %update-metadata
+      ?.  (~(has by synced) group-path.fact)  ~
+      (give group-path.fact fact)
+    ::
+        %remove
+      ?.  (~(has by synced) group-path.fact)  ~
+      (give group-path.fact fact)
+    ==
+  ::
+  ++  handle-foreign
+    ?+  -.fact  ~
+        %add
+      ?.  =(src.bowl (~(got by synced) group-path.fact))  ~
+      (poke fact)
+    ::
+        %update-metadata
+      ?.  =(src.bowl (~(got by synced) group-path.fact))  ~
+      (poke [%add +.fact])
+    ::
+        %remove
+      ?.  =(src.bowl (~(got by synced) group-path.fact))  ~
+      (poke fact)
+    ==
+  ::
+  ++  give
+    |=  [=path upd=metadata-update]
+    ^-  (list card)
+    [%give %fact ~[[%group path]] %metadata-update !>(upd)]~
+  ::
+  ++  poke
+    |=  act=metadata-action
+    ^-  (list card)
+    [%pass / %agent [our.bowl %metadata-store] %poke %metadata-action !>(act)]~
+  --
+::
+++  kick
+  |=  wir=wire
+  ^-  (quip card _state)
+  :_  state
+  ?+  wir  !!
+      [%updates ~]
+    [%pass /updates %agent [our.bowl %metadata-store] %watch /updates]~
+  ::
+      [%group @ *]
+    ?.  (~(has by synced) t.wir)  ~
+    =/  =ship  (~(got by synced) t.wir)
+    ?:  =(ship our.bowl)
+      [%pass wir %agent [our.bowl %metadata-store] %watch wir]~
+    [%pass wir %agent [ship %metadata-hook] %watch wir]~
```

Comment:
Can't you use `[ship %metadata-hook]` in both these cases? Doesn't look like there's a need for the check here.

Reply:
No, because if we're subscribing to a path on our ship, then we want to subscribe to our store.

Reply:
Durr, I hadn't seen the `-hook` `-store` difference. Disregard me!